pkg/pillar/cmd/volumemgr/handlecontainervolume.go

```diff
@@ -0,0 +1,282 @@
+package volumemgr
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"runtime"
+	"syscall"
+
+	"github.com/lf-edge/eve/pkg/pillar/containerd"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	// relative path to rootfs for an individual container
+	containerRootfsPath = "rootfs/"
+	// container config file name
+	imageConfigFilename = "image-config.json"
+	// contains conatiner's image name.
+	imageNameFilename = "image-name"
+)
+
+// prepareContainerVolume prepares a writable snapshot from an OCI layers bundle
+// We always do it from scratch all the way, ignoring any existing state
+// that may have accumulated (like existing snapshots being available, etc.)
+// This effectively voids any kind of caching, but on the flip side frees us
+// from cache invalidation. Additionally we deposit an OCI config json file and image name
+// next to the rootfs so that the effective structure becomes:
+//    rootPath/rootfs, rootPath/image-config.json, rootPath/image-name
+// We also expect rootPath to end in a basename that becomes containerd's snapshotID
+func prepareContainerVolume(ctx *volumemgrContext, rootPath string, reference string, rootBlobStatus *types.BlobStatus) error {
```

Comment:
Similar to the previous question. Should all of this be in here? This looks like a lot of containerd-specific detail?

Reply:
Moved.